presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConnectorConfig.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class ElasticsearchConnectorConfig
+{
+    /**
+     * The schema name to use in the connector.
+     */
+    private String defaultSchema = "default";
+
+    /**
+     * Set of tables known to this connector. For each table, a description file may be present in the catalog folder which describes columns for the given topic.
+     */
+    private Set<String> tableNames = ImmutableSet.of();
+
+    /**
+     * Folder holding the JSON description files for Elasticsearch Index.
+     */
+    private File tableDescriptionDir = new File("etc/elasticsearch/");
+
+    /**
+     * Scroll batch size.
+     */
+    private int scrollSize = 1000;
+
+    /**
+     * Scroll timeout.
+     */
+    private Duration scrollTime = new Duration(60, TimeUnit.SECONDS);
+
+    /**
+     * Max number of hits per Elasticsearch request could fetch.
+     */
+    private int maxHits = 1000000;
+
+    /**
+     * Elasticsearch request timeout.
+     */
+    private Duration requestTimeout = new Duration(60, TimeUnit.SECONDS);
+
+    @NotNull
+    public File getTableDescriptionDir()
+    {
+        return tableDescriptionDir;
+    }
+
+    @Config("elasticsearch.table-description-dir")
+    public ElasticsearchConnectorConfig setTableDescriptionDir(File tableDescriptionDir)
+    {
+        this.tableDescriptionDir = tableDescriptionDir;
+        return this;
+    }
+
+    @NotNull
+    public Set<String> getTableNames()
+    {
+        return tableNames;
+    }
+
+    @Config("elasticsearch.table-names")
+    public ElasticsearchConnectorConfig setTableNames(String tableNames)
+    {
+        this.tableNames = ImmutableSet.copyOf(Splitter.on(',').omitEmptyStrings().trimResults().split(tableNames));
+        return this;
+    }
+
+    @NotNull
+    public String getDefaultSchema()
+    {
+        return defaultSchema;
+    }
+
+    @Config("elasticsearch.default-schema")
+    public ElasticsearchConnectorConfig setDefaultSchema(String defaultSchema)
+    {
+        this.defaultSchema = defaultSchema;
+        return this;
+    }
+
+    @NotNull
+    public int getScrollSize()
+    {
+        return scrollSize;
+    }
+
+    @Config("elasticsearch.scroll-size")
+    public ElasticsearchConnectorConfig setScrollSize(int scrollSize)
+    {
+        this.scrollSize = scrollSize;
+        return this;
+    }
+
+    @NotNull
+    public Duration getScrollTime()
+    {
+        return scrollTime;
+    }
+
+    @Config("elasticsearch.scroll-time")
+    public ElasticsearchConnectorConfig setScrollTime(Duration scrollTime)
+    {
+        this.scrollTime = scrollTime;
+        return this;
+    }
+
+    @NotNull
+    public int getMaxHits()
+    {
+        return maxHits;
+    }
+
+    @Config("elasticsearch.max-hits")
+    public ElasticsearchConnectorConfig setMaxHits(int maxHits)
+    {
+        this.maxHits = maxHits;
+        return this;
+    }
+
+    @NotNull
+    public Duration getRequestTimeout()
```

Comment:
What happens when request timeout is `0`? Please take a look to see whether we need to enforce a `@MinDuration("...")` constraint.